ui/raidboss/timeline_parser.ts

```diff
@@ -395,7 +395,7 @@ export class TimelineParser {
     replaceLang: Lang,
     isGlobal: boolean,
   ): string {
-    if (!this.replacements)
+    if (this.replacements.length === 0)
```

Comment:
I think this is an incorrect logic change.  An empty array is truthy, and if you early out here you won't get the common replacements below.

Reply:
So then this check literally did nothing and can be removed entirely? Because `this.replacements` isn't nullable/optional, it's always an array of some sort, so the condition always evaluates to false?